lib/ember-test-helpers/abstract-test-module.js

```diff
@@ -100,12 +104,26 @@ export default Klass.extend({
   },
 
   setupContext(options) {
-    var config = assign({
+    let context = this.getContext();
+
+    merge(context, {
       dispatcher: null,
       inject: {}
-    }, options);
+    });
+    merge(context, options);
+
+    setContext(context);
+    this.context = context;
+  },
+
+  setContext(context) {
+    this.context = context;
+  },
+
+  getContext() {
+    if (this.context) { return this.context; }
 
-    setContext(config);
+    return this.context = getContext() || {};
```

Comment:
Sure, seems functionally the same.  Out of curiosity what makes you prefer that pattern over what I did originally?